grug/app/src/app.rs

```diff
@@ -48,23 +51,25 @@ pub struct App<DB, VM, PP = NaiveProposalPreparer> {
     query_gas_limit: u64,
 }
 
-impl<DB, VM, PP> App<DB, VM, PP> {
-    pub fn new(db: DB, vm: VM, pp: PP, query_gas_limit: u64) -> Self {
+impl<DB, VM, PP, ID> App<DB, VM, PP, ID> {
+    pub fn new(db: DB, vm: VM, pp: PP, indexer: ID, query_gas_limit: u64) -> Self {
         Self {
             db,
             vm,
             pp,
+            indexer,
             query_gas_limit,
         }
     }
 }
 
-impl<DB, VM, PP> App<DB, VM, PP>
+impl<DB, VM, PP, ID> App<DB, VM, PP, ID>
 where
     DB: Db,
     VM: Vm + Clone,
+    ID: Indexer,
     PP: ProposalPreparer,
```

Comment:
Let's put `ID` under `PP`. Let's keep the order of trait bound the same as the order of generics.